### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by sjchmiela**
https://github.com/Shopify/graphql-batch/issues/40#issuecomment-270868706

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,6 +48,19 @@ end
...
+```ruby
+MySchema = GraphQL::Schema.define do
+  query MyQueryType
+  
+  lazy_resolve(Promise, :sync)
+  instrument(:query, GraphQL::Batch::Setup)
+end
+```
+
+- When running `graphql` gem versions `< 1.3.0`:
```
Comment (dylanahsmith) [inline README.md]: I would prefer to just recommend using lazy_resolve and deprecating the old way of doing it.

I have opened https://github.com/Shopify/graphql-batch/pull/42 to deprecate the old way, although I think I will need to also add a warning for when an old version of the graphql gem is used.

File: README.md [hunk 2]
```diff
@@ -48,6 +48,19 @@ end
 
 Use the batch execution strategy with your schema
```
Comment (dylanahsmith) [inline README.md]: This sentence is describing the old way changing the execution strategy, which isn't applicable to the lazy_resolve approach.

File: README.md (hunk 1 above)
Comment (sjchmiela) [inline README.md]: Removed deprecated way

File: README.md (hunk 2 above)
Comment (sjchmiela) [inline README.md]: Changed to:
> Use lazy execution and instrumentation by `GraphQL::Batch` in your schema

**Pull request merged**